scan/lib/scan/xcpretty_reporter_options_generator.rb

```diff
@@ -55,7 +55,8 @@ def generate_reporter_options
       # adds another junit reporter in case the user does not specify one
       # this will be used to generate a results table and then discarded
       require 'tempfile'
-      Scan.cache[:temp_junit_report] = Tempfile.new("junit_report").path
+      @temp_junit_report = Tempfile.new("junit_report")
```

Comment:
I find it very convoluted to have to keep references all the way because we have to track a resource that is allocated down the call stack. I prefer when resources lifecycle is more explicit.